lxd/instance/drivers/qmp/commands.go

```diff
@@ -247,6 +249,87 @@ func (m *Monitor) SetMemoryBalloonSizeBytes(sizeBytes int64) error {
 	return m.run("balloon", args, nil)
 }
 
+// AddBlockDevice adds a block device.
+func (m *Monitor) AddBlockDevice(blockDev map[string]interface{}, device map[string]string) error {
+	revert := revert.New()
+	defer revert.Fail()
+
+	if blockDev != nil {
+		err := m.run("blockdev-add", blockDev, nil)
+		if err != nil {
+			return fmt.Errorf("Failed adding block device: %w", err)
+		}
+
+		revert.Add(func() {
+			blockDevDel := map[string]interface{}{
+				"node-name": blockDev["devName"],
+			}
+
+			m.run("blockdev-del", blockDevDel, nil)
+		})
+	}
+
+	if device != nil {
+		err := m.run("device_add", device, nil)
+		if err != nil {
+			return fmt.Errorf("Failed adding device: %w", err)
+		}
+	}
+
+	revert.Success()
+	return nil
+}
+
+// RemoveBlockDevice removes a block device.
+func (m *Monitor) RemoveBlockDevice(blockDevName string) error {
+	if blockDevName != "" {
+		blockDevName := map[string]string{
+			"node-name": blockDevName,
+		}
+
+		// Retry a few times in case the blockdev is in use.
+		err := m.run("blockdev-del", blockDevName, nil)
+		if err != nil {
+			if strings.Contains(err.Error(), "is in use") {
+				return api.StatusErrorf(http.StatusLocked, err.Error())
+			}
+
+			if strings.Contains(err.Error(), "Failed to find") {
+				return api.StatusErrorf(http.StatusNotFound, err.Error())
+			}
+
+			return fmt.Errorf("Failed removing block device: %w", err)
+		}
+	}
+
+	return nil
+}
+
+// RemoveDevice removes a device.
+func (m *Monitor) RemoveDevice(deviceID string) error {
+	// Check if disconnected
+	if m.disconnected {
+		return ErrMonitorDisconnect
+	}
+
+	if deviceID != "" {
+		deviceID := map[string]string{
+			"id": deviceID,
+		}
+
+		err := m.run("device_del", deviceID, nil)
+		if err != nil {
+			if strings.Contains(err.Error(), "not found") {
+				return api.StatusErrorf(http.StatusNotFound, err.Error())
```

Comment:
We can just return nil here right? (the device isn't present, so its removed already, nothing to do).